includes/CargoSQLQuery.php

```diff
@@ -1024,18 +1028,9 @@ private function handleVirtualFields() {
 		// joins - match the order in $this->mAliasedTableNames to the
 		// order of the tables within the joins.
 		if ( count( $this->mAliasedTableNames ) > 1 ) {
-			$orderedTableAliases = [];
-			foreach ( $this->mCargoJoinConds as $joinCond ) {
-				$table1 = $joinCond['table1'];
-				$table2 = $joinCond['table2'];
-				if ( !in_array( $table1, $orderedTableAliases ) ) {
-					$orderedTableAliases[] = $table1;
-				}
-				if ( !in_array( $table2, $orderedTableAliases ) ) {
-					$orderedTableAliases[] = $table2;
-				}
-			}
-
+			// Fandom-start. PLATFORM-10866: Reorder tables to join tables in correct order
+			$orderedTableAliases = $this->getOrderedTables();
+			// Fandom-end
 			uksort( $this->mAliasedTableNames, static function ( $key1, $key2 ) use ( $orderedTableAliases ) {
 				return ( array_search( $key1, $orderedTableAliases ) - array_search( $key2, $orderedTableAliases ) );
 			} );
```

Comment:
Note: This code is new, on mw139 there was some sorting code. https://github.com/Wikia/mediawiki-extensions-Cargo/blob/81e9e5364c888495ba0899f3046e2c93126ff5fb/includes/CargoSQLQuery.php#L345-L352 
AFAIK joins are glued in table order inside the MW `SQLPlatform` class https://github.com/Wikia/mediawiki/blob/edcbec2f6842dce23b5ce4fcea4105c7a1cfdec5/includes/libs/rdbms/platform/SQLPlatform.php#L902